frontend/lib/decklist-parser.ts

```diff
@@ -0,0 +1,96 @@
+import lookup from "@/lib/api/spellbook-api";
+
+import type { FormattedApiResponse } from "@/lib/api/types";
+
+type CombosInDecklist = {
+  combosInDecklist: FormattedApiResponse[];
+  potentialCombos: FormattedApiResponse[];
+};
+
+type Deck = {
+  cards: string[];
+  numberOfCards: number;
+};
+
+export function convertDecklistToDeck(decklist: string): Deck {
+  let numberOfCardsInDeck = 0;
+
+  const cards = decklist
+    .split("\n")
+    .filter((cardName) => {
```

Comment:
I chose to do it as multiple array methods for ease of readability. Since the decklists are not likely to be greater than 100 entries, it's not really necessary to optimize it right now.